src/main/java/housing/Region.java

```diff
@@ -84,6 +130,51 @@ public void step() {
         // Stores regional household statistics after both regional markets have been cleared
         regionalHouseholdStats.record();
     }
+    
+    //******** Methods for Failed Bidders ******//   
+    public Map<Id<Household>,Integer> getFailedBidderCounter(){
+    	return this.failedBidderCounter;
+    }
+    
+    public void countFailedBidder(ArrayList<HouseBuyerRecord> failedBids){   
+    	for(HouseBuyerRecord failedBid:failedBids){
+    		Id<Household> householdId=failedBid.getBuyer().getHouseholdId();
+    		countFailedBidder(householdId);    		
+    	}
+    }
+    
+    public void countFailedBidder(Id<Household> householdId){   
+    	int cnt=0;
+    	if(failedBidderCounter.containsKey(householdId)){
+    		cnt=failedBidderCounter.get(householdId)+1;
+    	}
+    	else{
+    		cnt=1;
+    	}
+    	failedBidderCounter.put(householdId, cnt);
+    }
+    
+    public void removeSuccessfulBidder(Id<Household> householdId){
+    	if(failedBidderCounter.containsKey(householdId)){
+    		failedBidderCounter.remove(householdId);
+    	}
+    }
+    /**
+     * it is assuemd that a household will start trying to bid in neighbouring regions after failling to get a offer in its present 
+     * region for a specfic number of steps that is maxNumOfStepsInPresentRegion
+     * @param householdId
+     * @param maxNumOfStepsInPresentRegion: the maximum number of steps that households can bid in the present region
+     */
+    public boolean decideToTryNeighbouringRegion(Id<Household> householdId,int maxNumOfStepsInPresentRegion){
```

Comment:
Again, it is very strange to have a behavioural decision stored in the Region class... in principle, all behavioural stuff should sit in households or householdBehaviour...

Reply:
Yes, I will remove this method to Household or HouseholdBehaviour instead.

Reply:
Great, thanks.